src/core/packages/elasticsearch/server-internal/src/elasticsearch_service.test.ts

```diff
@@ -224,243 +227,274 @@ describe('#setup', () => {
 
   it('esNodeVersionCompatibility$ only starts polling when subscribed to', async () => {
     const mockedClient = mockClusterClientInstance.asInternalUser;
-    mockedClient.nodes.info.mockImplementation(() =>
-      elasticsearchClientMock.createErrorTransportRequestPromise(new Error())
-    );
+    // mocking successful response becuase error responses would trigger retries and complicate the test
+    mockedClient.nodes.info.mockResolvedValue({
+      nodes: {},
+    } as any);
 
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(0);
 
     const setupContract = await elasticsearchService.setup(setupDeps);
 
+    // Polling starts immediately when setup subscribes
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(1);
 
     tick();
 
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
 
     await firstValueFrom(setupContract.esNodesCompatibility$);
-    expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2); // shares the last value
+    // Should reuse the cached value from the first subscription
+    expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
   });
 
   it('esNodeVersionCompatibility$ stops polling when unsubscribed from', async () => {
     const mockedClient = mockClusterClientInstance.asInternalUser;
-    mockedClient.nodes.info.mockImplementation(() =>
-      elasticsearchClientMock.createErrorTransportRequestPromise(new Error())
-    );
+    mockedClient.nodes.info.mockResolvedValue({
+      nodes: {},
+    } as any);
 
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(0);
 
     const setupContract = await elasticsearchService.setup(setupDeps);
 
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(1);
 
+    expect(mockedClient.nodes.info).toHaveBeenCalledTimes(1);
+
     await firstValueFrom(setupContract.esNodesCompatibility$);
 
     tick();
 
     expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
   });
-});
 
-describe('#start', () => {
-  it('throws if called before `setup`', async () => {
-    await expect(() => elasticsearchService.start()).rejects.toMatchInlineSnapshot(
-      `[Error: ElasticsearchService needs to be setup before calling start]`
-    );
-  });
-
-  it('returns elasticsearch client as a part of the contract', async () => {
-    await elasticsearchService.setup(setupDeps);
-    const startContract = await elasticsearchService.start();
-    const client = startContract.client;
-
-    expect(client.asInternalUser).toBe(mockClusterClientInstance.asInternalUser);
-  });
-
-  it('should log.error non-compatible nodes error', async () => {
-    const defaultMessage = {
-      isCompatible: true,
-      kibanaVersion: '8.0.0',
-      incompatibleNodes: [],
-      warningNodes: [],
-    };
-    const observable$ = new BehaviorSubject<NodesVersionCompatibility>(defaultMessage);
+  it('esNodeVersionCompatibility$ only emits error after retries have been exhausted', async () => {
```

Comment:
Explicitly ensure that the error only propagates after `healthCheckRetry` number of times. The test config uses `retry: 1` for simplicity.

Reply:
we already have tests covering `pollEsNodesVersion` so retesting this in the elasticsearch service is redundant